integration/oci_calls/oci_call_test.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will verify the arguments of
+# running or stopping a container matches
+# with the OCI calls
+
+set -e
+
+cidir=$(dirname "$0")
+source "${cidir}/../../lib/common.bash"
+
+# Save logs
+TMP_FILE=$(mktemp runtimelogs.XXXXX)
+# Environment variables
+IMAGE="busybox"
+PAYLOAD="tail -f /dev/null"
+NAME="testoci"
+
+function remove_tmp_file {
+	rm -rf $TMP_FILE
+}
+
+trap remove_tmp_file EXIT
+
+# Get time to filter the logs
+function get_time() {
+	start_time=$(date "+%F %H:%M:%S")
+}
+
+# Get log for a specific time
+function get_debug_logs() {
+	sudo journalctl -q --since "$start_time" -o cat -a -t ${RUNTIME} > ${TMP_FILE}
+}
+
+# Find the arguments or oci calls for a specific command
+function check_arguments() {
+	list_arguments=$(grep -o "arguments=[^ ]*" ${TMP_FILE} --color|cut -d= -f2-|tr -d '"'|tr -d "\\\\")
+
+	[ -n "$list_arguments" ] || die "arguments are empty"
+
+	# Check arguments vs oci calls
+	for i in "${oci_call[@]}"; do
+		echo "$list_arguments" | grep -w "$i" > /dev/null
+	done
+}
+
+function setup() {
+	clean_env
+
+	check_processes
+
+	extract_kata_env
+
+	# Enable full debug
+	sudo sed -i 's/#enable_debug = true/enable_debug = true/g' ${RUNTIME_CONFIG_PATH}
+}
+
+function run_oci_call() {
+	local -a oci_call=( "create" "state" "start" )
+
+	get_time
+
+	# Start a container
+	docker run -d --runtime=${RUNTIME} --name=${NAME} ${IMAGE} ${PAYLOAD}
+
+	get_debug_logs
+
+	check_arguments
+}
+
+function stop_oci_call() {
+	local -a oci_call=( "kill" "state" "delete" )
+
+	get_time
+
+	# Stop a container
+	docker stop ${NAME}
+
+	get_debug_logs
+
+	check_arguments
+
+	docker rm -f ${NAME} > /dev/null
+}
+
+function run_oci_call_true() {
+	# Find docker version
+	version=$(docker version --format '{{.Server.Version}}' | cut -d '.' -f1-2)
+	echo "$version>=18.06" | bc
```

Comment:
This is neat, but actually incorrect because the value you need to check on is not `$?` - it's the number that `bc` returns. So something like:

```
result=$(echo "$version>=18.06" | bc)
if [ $result -ne 1 ]; then
    ...
```